.github/workflows/build.yml

```diff
@@ -5,6 +5,10 @@ on:
   push:
     branches:
       - main
+# the most recent commit in the PR would cancel older commit's in-progress workflows
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref_name }}
```

Comment:
I don't understand the `github.ref_name` part. Can you clarify?

Reply:
The github.ref_name is the reference to the Git branch or tag name that initiated the workflow. According to how the workflow is triggered above, it can be triggered by pull request or main branch.  When it is triggered by the main branch's push event, the github.ref_name would become "main" branch.

Reply:
This leads me to think that maybe I should delete the github.ref_name. It may also cancel concurrent pushes to main, which is probably not what we want. Let me do this change.

Reply:
Thank you for noticing and asking about this github.ref_name! Let's see how it behaves :)